src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs

```diff
@@ -1393,5 +1393,17 @@ public void AddFaasAttribute(string name, object value)
             var faasAttrib = _attribDefs.GetFaasAttribute(name);
             TransactionMetadata.UserAndRequestAttributes.TrySetValue(faasAttrib, value);
         }
+
+        public void AddCloudSdkAttribute(string name, object value)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                Log.Debug($"AddCloudSdkAttribute - Unable to set Cloud value on transaction because the key is null/empty");
```

Comment:
Should this say `name` instead of `key`?

Reply:
I was following the pattern in `AddLambdaAttribute` and `AddFaasAttribute` but yeah, "name" is probably clearer.